### PR TITLE
Bring README up to date and with more details\TESTING=none

### PR DESCRIPTION
**Pull request by scottmcmaster**
Please review the content for completeness/accuracy, I'm sure there's a lot to be improved here. Thanks!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,51 +2,170 @@
...
-3. `bundle`
-4. `irb -Ilib -rpromoted/ruby/client`
+### LogRequest
 
-A console will launch with the library loaded.  Here is example code to use.
+Output of ```prepare_for_logging``` as well as an ouput of an SDK call to ```deliver```, input to ```send_log_request``` to log to Promoted
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:request``` | Request | No | The underlying request for content to log.
+```:insertion``` | [] of Insertion | No | The insertions, which are either the original request insertions or the insertions resulting from a call to ```deliver``` if such call occurred.
```
Comment (danbosnichill) [inline README.md:140]: We chatted in person.  It'd be good to document the motivation and benefit of the clients.

**Review by danbosnichill — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -2,51 +2,170 @@
...
 
-This version of the library only supports preparing objects for logging.  TODO - support Delivery API.
+More information at [http://www.promoted.ai](http://www.promoted.ai)
 
-## Expected pseudo-code flow for Metrics logging
+## Installation
+```gem 'promoted-ruby-client'```
 
-This example is for the integration where we do not want to modify the list of items to include `insertionId`.  TODO - add this example too.
+## Local Development
```
[the remark this answers is not in the document]
Comment (scottmcmaster) [reply README.md:10]: This is for anyone who would like to try out the client in the repl.

File: README.md [hunk 3]
```diff
@@ -2,51 +2,170 @@
...
-  # Send JSON to Metrics API.
-  log_to_promoted_event_api(log_request)
-end
+This client will suffice for building log requests. To send actually send traffing to the API, some configuration is required.
+
+```rb
+client = Promoted::Ruby::Client::PromotedClient.new({
+  :metrics_endpoint = "https://<get this from Promoted>",
+  :delivery_endpoint = "https://<get this from Promoted>",
+  :api_key = "<get this from Promoted>"
```
[the remark this answers is not in the document]
Comment (scottmcmaster) [reply README.md]: Fixed

File: README.md [hunk 4]
```diff
@@ -2,51 +2,170 @@
...
 
-The `prepare_for_logging` call assumes the client has already handled pagination.  It needs a `Request.paging.offset` to be passed in for the number of items deep that the page is.
+---
 
-## Example to run the client
+### Timing
+Request timing information.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:client_log_timestamp``` | Number | Yes (will be filled in if not provided) | Time that an event was created.
```
[the remark this answers is not in the document]
Comment (scottmcmaster) [reply README.md:69]: Removed timing. Would note that the way the SDKs are currently written, clients can provide/override these.

File: README.md [hunk 5]
```diff
@@ -2,51 +2,170 @@
...
+---
+### CohortMembership
+Useful fields for experimentation during the delivery phase.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:user_info``` | UserInfo | Yes | The user info structure.
+```:arm``` | String | Yes | 'CONTROL' or one of the TREATMENT values (see [constants.rb](https://github.com/promotedai/promoted-ruby-client/blob/main/lib/promoted/ruby/client/constants.rb)).
```
[the remark this answers is not in the document]
Comment (scottmcmaster) [reply README.md:76]: I would definitely plan to add some good experiment examples that would drive this issue.

File: README.md [hunk 6]
```diff
@@ -2,51 +2,170 @@
...
+### Insertion
+Content being served at a certain position.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:user_info``` | UserInfo | Yes | The user info structure.
+```:insertion_id``` | String | Yes | Generated by the SDK (*do not set*)
+```:request_id``` | String | Yes | Generated by the SDK (*do not set*)
+```:content_id``` | String | No | Identifier for the content to be shown, must be set.
+```:properties``` | Properties | Yes | Any additional custom properties to associate. For v1 integrations, it is fine not to fill in all the properties.
+```:position``` | Number | Yes | Content location in the list of insertions, calculated by Promoted and uses the Paging parameters. (*do not set*)
```
[the remark this answers is not in the document]
Comment (scottmcmaster) [reply README.md:101]: Done.

File: README.md (hunk 1 above)
Comment (scottmcmaster) [reply README.md:140]: Agree. Let's chat offline about the best format/venue for that. I will add it to the issue I'm tracking.

File: README.md [hunk 7]
```diff
@@ -2,51 +2,165 @@
...
+```:request``` | Request | No | The underlying request for content to log.
+```:insertion``` | [] of Insertion | No | The insertions, which are either the original request insertions or the insertions resulting from a call to ```deliver``` if such call occurred.
+---
+
+### ClientResponse
+Output of ```deliver```, includes the insertions as well as a suitable ```LogRequest``` for forwarding to Metrics API.
+Field Name | Type | Optional? | Description
+---------- | ---- | --------- | -----------
+```:insertion``` | [] of Insertion | No | The insertions, which are from Delivery API (when ```deliver``` was called, i.e. we weren't either only-log or part of an experiment) or the input insertions (when the other conditions don't hold).
+```:log_request``` | LogRequest | Yes | A message suitable for logging to Metrics API via ```send_log_request```. If the call to ```deliver``` was made (i.e. the request was not part of the CONTROL arm of an experiment or marked to only log), ```:log_request``` will not be set, as you can assume logging was performed on the server-side by Promoted.
```
Comment (scottmcmaster) [inline README.md:141]: @danbosnichill I clarified the circumstances under which log_request will be set on the client response, please feel free to double-check.
